html/modules/custom/reliefweb_entities/src/Entity/Source.php

```diff
@@ -207,27 +207,47 @@ protected function getOrganizationDetails() {
   protected function getOrganizationSocialMediaLinks() {
     $links = [];
 
-    if (!$this->field_links->isEmpty()) {
-      // Get the list of allowed social media.
-      $allowed = \Drupal::config('reliefweb_entities.settings')
-        ->get('allowed_social_media_links') ?? [];
-
-      // Prepare the links.
-      foreach ($this->field_links as $link) {
-        $host = parse_url($link->uri, PHP_URL_HOST);
-        foreach ($allowed as $key => $name) {
-          if ($host && mb_strpos($host, $key) !== FALSE) {
-            $links[$key] = [
-              'url' => UrlHelper::encodeUrl($link->uri),
-              'title' => $name,
-            ];
-            break;
-          }
+    if ($this->field_links->isEmpty()) {
+      return [];
+    }
+
+    // Get the list of allowed social media.
+    $allowed = \Drupal::config('reliefweb_entities.settings')
+      ->get('allowed_social_media_links') ?? [];
+
+    if (empty($allowed)) {
+      return [];
+    }
+
+    // Build array of allowed domains.
+    $allowed_domains = [];
+    foreach ($allowed as $domain => $name) {
+      $allowed_domains[str_replace('_', '.', $domain)] = $name;
+    }
+
+    // Prepare the links.
+    foreach ($this->field_links as $link) {
+      $host = parse_url($link->uri, PHP_URL_HOST);
+      if (!$host) {
+        continue;
+      }
+
+      foreach ($allowed_domains as $domain => $name) {
+        if (mb_strpos($host, $domain) !== FALSE) {
+          $links[$name] = [
+            'url' => UrlHelper::encodeUrl($link->uri),
+            'title' => $name,
+          ];
+          break;
         }
       }
     }
 
     if (!empty($links)) {
+      uasort($links, function ($a, $b) {
+        return strcmp($a['title'], $b['title']);
+      });
```

Comment:
I would suggest to use the collated sort from the LocalizationHelper: https://github.com/UN-OCHA/rwint9-site/blob/develop/html/modules/custom/reliefweb_utility/src/Helpers/LocalizationHelper.php#L82 to ensure it's in "natural order" and language aware.

```
LocalizationHelper::collatedSort($links, 'title');
```

Reply:
I made the change.